Java 8u121 JavaDoc workaround is not working

Environment:

```
Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-10T18:41:47+02:00)
Maven home: C:\Program Files\Java\apache-maven-3.3.9
Java version: 1.8.0_121, vendor: Oracle Corporation
Java home: C:\Program Files\Java\jdk1.8.0_121\jre
Default locale: en_US, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "dos"
```

Error

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 42.502 s
[INFO] Finished at: 2017-03-21T05:59:33+02:00
[INFO] Final Memory: 26M/131M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:2.9.1:jar (attach-javadocs) on project eddsa: MavenReportException: Error while creating archive:
[ERROR] Exit code: 1 - javadoc: error - Argument for -header contains JavaScript.
[ERROR] Use --allow-script-in-comments to allow use of JavaScript.
[ERROR]
[ERROR] Command line was: "C:\Program Files\Java\jdk1.8.0_121\jre\..\bin\javadoc.exe" @options @packages
[ERROR]
[ERROR] Refer to the generated Javadoc files in 'D:\Projects\apache\str4d\ed25519-java\target\apidocs' dir.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

The fix is simple - see https://github.com/neo4j/neo4j-java-driver/pull/318. However, the build still fails due to malformed Javadoc comments:

```
[ERROR] Exit code: 1 - D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:324: warning: no
@param for data
[ERROR] public byte[] signOneShot(byte[] data) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:324: warning: no @return
[ERROR] public byte[] signOneShot(byte[] data) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:342: warning: no @param for data
[ERROR] public byte[] signOneShot(byte[] data, int off, int len) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:342: warning: no @param for off
[ERROR] public byte[] signOneShot(byte[] data, int off, int len) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:342: warning: no @param for len
[ERROR] public byte[] signOneShot(byte[] data, int off, int len) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:342: warning: no @return
[ERROR] public byte[] signOneShot(byte[] data, int off, int len) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:362: warning: no @param for data
[ERROR] public boolean verifyOneShot(byte[] data, byte[] signature) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:362: warning: no @param for signature
[ERROR] public boolean verifyOneShot(byte[] data, byte[] signature) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:362: warning: no @return
[ERROR] public boolean verifyOneShot(byte[] data, byte[] signature) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:380: warning: no @param for data
[ERROR] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:380: warning: no @param for off
[ERROR] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:380: warning: no @param for len
[ERROR] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:380: warning: no @param for signature
[ERROR] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:380: warning: no @return
[ERROR] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:398: warning: no @param for data
[ERROR] public boolean verifyOneShot(byte[] data, byte[] signature, int sigoff, int siglen) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:398: warning: no @param for signature
[ERROR] public boolean verifyOneShot(byte[] data, byte[] signature, int sigoff, int siglen) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:398: warning: no @param for sigoff
[ERROR] public boolean verifyOneShot(byte[] data, byte[] signature, int sigoff, int siglen) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:398: warning: no @param for siglen
[ERROR] public boolean verifyOneShot(byte[] data, byte[] signature, int sigoff, int siglen) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:398: warning: no @return
[ERROR] public boolean verifyOneShot(byte[] data, byte[] signature, int sigoff, int siglen) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:416: warning: no @param for data
[ERROR] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature, int sigoff, int siglen) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:416: warning: no @param for off
[ERROR] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature, int sigoff, int siglen) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:416: warning: no @param for len
[ERROR] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature, int sigoff, int siglen) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:416: warning: no @param for signature
[ERROR] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature, int sigoff, int siglen) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:416: warning: no @param for sigoff
[ERROR] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature, int sigoff, int siglen) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:416: warning: no @param for siglen
[ERROR] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature, int sigoff, int siglen) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:416: warning: no @return
[ERROR] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature, int sigoff, int siglen) throws SignatureException {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:438: error: element not closed:
a
[ERROR] * @deprecated replaced with <a href="#engineSetParameter(java.security.spec.AlgorithmParameterSpec)">
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAKey.java:30: warning: no @return
[ERROR] EdDSAParameterSpec getParams();
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\Utils.java:26: warning: no @param for b
[ERROR] public static int equal(int b, int c) {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\Utils.java:26: warning: no @param for c
[ERROR] public static int equal(int b, int c) {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\Utils.java:39: warning: no @param for b
[ERROR] public static int equal(byte[] b, byte[] c) {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\Utils.java:39: warning: no @param for c
[ERROR] public static int equal(byte[] b, byte[] c) {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\bigint\BigIntegerLittleEndianEncoding.java:45: warning: no @param for x
[ERROR] public byte[] encode(BigInteger x) {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\bigint\BigIntegerLittleEndianEncoding.java:79: warning: no @param for in
[ERROR] public BigInteger toBigInteger(byte[] in) {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\bigint\BigIntegerLittleEndianEncoding.java:79: warning: no @return
[ERROR] public BigInteger toBigInteger(byte[] in) {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\Encoding.java:33: warning: no @param for x
[ERROR] public abstract byte[] encode(FieldElement x);
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\Encoding.java:51: warning: no @param for x
[ERROR] public abstract boolean isNegative(FieldElement x);
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ScalarOps.java:21: warning: no description for @param
[ERROR] * @param s
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ScalarOps.java:28: warning: no description for @param
[ERROR] * @param a
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ScalarOps.java:29: warning: no description for @param
[ERROR] * @param b
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ScalarOps.java:30: warning: no description for @param
[ERROR] * @param c
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\GroupElement.java:506: warning: empty <p> tag
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\GroupElement.java:515: warning: empty <p> tag
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\GroupElement.java:648: warning: empty <p> tag
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\GroupElement.java:655: warning: empty <p> tag
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\GroupElement.java:867: error: malformed HTML
[ERROR] *   $a[31] <= 127$
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\GroupElement.java:39: warning: empty <p> tag
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\GroupElement.java:39: error: element not closed: ul
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\GroupElement.java:226: warning: empty <p> tag
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\GroupElement.java:226: error: element not closed: ul
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ed25519\Ed25519FieldElement.java:65: warning: empty <p> tag
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ed25519\Ed25519FieldElement.java:70: warning: empty <p> tag
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ed25519\Ed25519FieldElement.java:70: error:
element not closed: ul
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ed25519\Ed25519FieldElement.java:93: warning: empty <p> tag
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ed25519\Ed25519FieldElement.java:98: warning: empty <p> tag
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ed25519\Ed25519FieldElement.java:98: error:
element not closed: ul
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ed25519\Ed25519FieldElement.java:119: warning: empty <p> tag
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ed25519\Ed25519FieldElement.java:123: warning: empty <p> tag
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ed25519\Ed25519FieldElement.java:123: error: element not closed: ul
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ed25519\Ed25519FieldElement.java:142: warning: empty <p> tag
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ed25519\Ed25519FieldElement.java:149: warning: empty <p> tag
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ed25519\Ed25519FieldElement.java:393: warning: empty <p> tag
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ed25519\Ed25519FieldElement.java:397: warning: empty <p> tag
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ed25519\Ed25519FieldElement.java:549: warning: empty <p> tag
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ed25519\Ed25519FieldElement.java:553: warning: empty <p> tag
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ed25519\Ed25519LittleEndianEncoding.java:29: warning: empty <p> tag
[ERROR] * <p>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ed25519\Ed25519LittleEndianEncoding.java:31: warning: empty <p> tag
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ed25519\Ed25519LittleEndianEncoding.java:37: warning: empty <p> tag
[ERROR] * <p>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ed25519\Ed25519LittleEndianEncoding.java:245: warning: empty <p> tag
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ed25519\Ed25519LittleEndianEncoding.java:245: error: element not closed: ul
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ed25519\Ed25519ScalarOps.java:330: warning:
empty <p> tag
[ERROR] * <p><ul>
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\spec\EdDSAParameterSpec.java:39: warning: no @param for curve
[ERROR] public EdDSAParameterSpec(Curve curve, String hashAlgo,
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\spec\EdDSAParameterSpec.java:39: warning: no @param for hashAlgo
[ERROR] public EdDSAParameterSpec(Curve curve, String hashAlgo,
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\spec\EdDSAParameterSpec.java:39: warning: no @param for sc
[ERROR] public EdDSAParameterSpec(Curve curve, String hashAlgo,
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\spec\EdDSAParameterSpec.java:39: warning: no @param for B
[ERROR] public EdDSAParameterSpec(Curve curve, String hashAlgo,
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\spec\EdDSAPrivateKeySpec.java:35: warning: no @param for seed
[ERROR] public EdDSAPrivateKeySpec(byte[] seed, EdDSAParameterSpec spec) {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\spec\EdDSAPrivateKeySpec.java:35: warning: no @param for spec
[ERROR] public EdDSAPrivateKeySpec(byte[] seed, EdDSAParameterSpec spec) {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\spec\EdDSAPrivateKeySpec.java:70: warning: no @param for spec
[ERROR] public EdDSAPrivateKeySpec(EdDSAParameterSpec spec, byte[] h) {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\spec\EdDSAPublicKeySpec.java:30: warning: no @param for pk
[ERROR] public EdDSAPublicKeySpec(byte[] pk, EdDSAParameterSpec spec) {
[ERROR] ^
[ERROR] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\spec\EdDSAPublicKeySpec.java:30: warning: no @param for spec
[ERROR] public EdDSAPublicKeySpec(byte[] pk, EdDSAParameterSpec spec) {
[ERROR] ^
[ERROR]
[ERROR] Command line was: "C:\Program Files\Java\jdk1.8.0_121\jre\..\bin\javadoc.exe" @options @packages
[ERROR]
[ERROR] Refer to the generated Javadoc files in 'D:\Projects\apache\str4d\ed25519-java\target\apidocs' dir.
```

Not yet:
```
[INFO] --- maven-javadoc-plugin:2.9.1:jar (attach-javadocs) @ eddsa ---
[INFO]
Loading source files for package net.i2p.crypto.eddsa...
Loading source files for package net.i2p.crypto.eddsa.math.bigint...
Loading source files for package net.i2p.crypto.eddsa.math...
Loading source files for package net.i2p.crypto.eddsa.math.ed25519...
Loading source files for package net.i2p.crypto.eddsa.spec...
Constructing Javadoc information...
1 error
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 29.941 s
[INFO] Finished at: 2017-04-01T19:19:21+03:00
[INFO] Final Memory: 27M/158M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:2.9.1:jar (attach-javadocs) on project eddsa: MavenReportException: Error while creating archive:
[ERROR] Exit code: 1 - javadoc: error - Argument for -header contains JavaScript.
[ERROR] Use --allow-script-in-comments to allow use of JavaScript.
```

After commenting out the javascript `<header>` part (adding the ` --allow-script-in-comments` did not help) there are warnings (see below) but no errors and the plugin succeeds (except for the javascript part...)

```
WARNING] Javadoc Warnings
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:324: warning: no @param for data
[WARNING] public byte[] signOneShot(byte[] data) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:324: warning: no @return
[WARNING] public byte[] signOneShot(byte[] data) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:342: warning: no @param for data
[WARNING] public byte[] signOneShot(byte[] data, int off, int len) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:342: warning: no @param for off
[WARNING] public byte[] signOneShot(byte[] data, int off, int len) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:342: warning: no @param for len
[WARNING] public byte[] signOneShot(byte[] data, int off, int len) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:342: warning: no @return
[WARNING] public byte[] signOneShot(byte[] data, int off, int len) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:362: warning: no @param for data
[WARNING] public boolean verifyOneShot(byte[] data, byte[] signature) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:362: warning: no @param for signature
[WARNING] public boolean verifyOneShot(byte[] data, byte[] signature) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:362: warning: no @return
[WARNING] public boolean verifyOneShot(byte[] data, byte[] signature) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:380: warning: no @param for data
[WARNING] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:380: warning: no @param for off
[WARNING] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:380: warning: no @param for len
[WARNING] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:380: warning: no @param for signature
[WARNING] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:380: warning: no @return
[WARNING] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:398: warning: no @param for data
[WARNING] public boolean verifyOneShot(byte[] data, byte[] signature, int sigoff, int siglen) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:398: warning: no @param for signature
[WARNING] public boolean verifyOneShot(byte[] data, byte[] signature, int sigoff, int siglen) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:398: warning: no @param for sigoff
[WARNING] public boolean verifyOneShot(byte[] data, byte[] signature, int sigoff, int siglen) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:398: warning: no @param for siglen
[WARNING] public boolean verifyOneShot(byte[] data, byte[] signature, int sigoff, int siglen) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:398: warning: no @return
[WARNING] public boolean verifyOneShot(byte[] data, byte[] signature, int sigoff, int siglen) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:416: warning: no @param for data
[WARNING] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature, int sigoff, int siglen) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:416: warning: no @param for off
[WARNING] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature, int sigoff, int siglen) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:416: warning: no @param for len
[WARNING] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature, int sigoff, int siglen) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:416: warning: no @param for signature
[WARNING] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature, int sigoff, int siglen) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:416: warning: no @param for sigoff
[WARNING] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature, int sigoff, int siglen) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:416: warning: no @param for siglen
[WARNING] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature, int sigoff, int siglen) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAEngine.java:416: warning: no @return
[WARNING] public boolean verifyOneShot(byte[] data, int off, int len, byte[] signature, int sigoff, int siglen) throws SignatureException {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\EdDSAKey.java:30: warning: no @return
[WARNING] EdDSAParameterSpec getParams();
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\Utils.java:26: warning: no @param for b
[WARNING] public static int equal(int b, int c) {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\Utils.java:26: warning: no @param for c
[WARNING] public static int equal(int b, int c) {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\Utils.java:39: warning: no @param for b
[WARNING] public static int equal(byte[] b, byte[] c) {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\Utils.java:39: warning: no @param for c
[WARNING] public static int equal(byte[] b, byte[] c) {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\bigint\BigIntegerLittleEndianEncoding.java:45: warning: no @param for x
[WARNING] public byte[] encode(BigInteger x) {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\bigint\BigIntegerLittleEndianEncoding.java:79: warning: no @param for in
[WARNING] public BigInteger toBigInteger(byte[] in) {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\bigint\BigIntegerLittleEndianEncoding.java:79: warning: no @return
[WARNING] public BigInteger toBigInteger(byte[] in) {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\Encoding.java:33: warning: no @param for x
[WARNING] public abstract byte[] encode(FieldElement x);
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\Encoding.java:51: warning: no @param for x
[WARNING] public abstract boolean isNegative(FieldElement x);
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ScalarOps.java:21: warning: no description for @param
[WARNING] * @param s
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ScalarOps.java:28: warning: no description for @param
[WARNING] * @param a
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ScalarOps.java:29: warning: no description for @param
[WARNING] * @param b
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\ScalarOps.java:30: warning: no description for @param
[WARNING] * @param c
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\GroupElement.java:508: warning: empty <p> tag
[WARNING] * <p><ul>
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\GroupElement.java:517: warning: empty <p> tag
[WARNING] * <p><ul>
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\GroupElement.java:650: warning: empty <p> tag
[WARNING] * <p><ul>
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\math\GroupElement.java:657: warning: empty <p> tag
[WARNING] * <p><ul>
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\spec\EdDSAParameterSpec.java:39: warning: no @param for curve
[WARNING] public EdDSAParameterSpec(Curve curve, String hashAlgo,
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\spec\EdDSAParameterSpec.java:39: warning: no @param for hashAlgo
[WARNING] public EdDSAParameterSpec(Curve curve, String hashAlgo,
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\spec\EdDSAParameterSpec.java:39: warning: no @param for sc
[WARNING] public EdDSAParameterSpec(Curve curve, String hashAlgo,
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\spec\EdDSAParameterSpec.java:39: warning: no @param for B
[WARNING] public EdDSAParameterSpec(Curve curve, String hashAlgo,
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\spec\EdDSAPrivateKeySpec.java:35: warning: no @param for seed
[WARNING] public EdDSAPrivateKeySpec(byte[] seed, EdDSAParameterSpec spec) {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\spec\EdDSAPrivateKeySpec.java:35: warning: no @param for spec
[WARNING] public EdDSAPrivateKeySpec(byte[] seed, EdDSAParameterSpec spec) {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\spec\EdDSAPrivateKeySpec.java:70: warning: no @param for spec
[WARNING] public EdDSAPrivateKeySpec(EdDSAParameterSpec spec, byte[] h) {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\spec\EdDSAPublicKeySpec.java:30: warning: no @param for pk
[WARNING] public EdDSAPublicKeySpec(byte[] pk, EdDSAParameterSpec spec) {
[WARNING] ^
[WARNING] D:\Projects\apache\str4d\ed25519-java\src\net\i2p\crypto\eddsa\spec\EdDSAPublicKeySpec.java:30: warning: no @param for spec
[WARNING] public EdDSAPublicKeySpec(byte[] pk, EdDSAParameterSpec spec) {
[WARNING] ^
```

--- a/pom.xml
+++ b/pom.xml
@@ -98,6 +98,9 @@
         <artifactId>maven-javadoc-plugin</artifactId>
         <version>2.9.1</version>
         <configuration>
+          <charset>UTF-8</charset>
+          <docencoding>UTF-8</docencoding>
+          <encoding>UTF-8</encoding>
           <header>&lt;script type='text/x-mathjax-config'&gt;
   MathJax.Hub.Config({
     tex2jax: {

--- a/pom.xml
+++ b/pom.xml
@@ -107,6 +107,8 @@
   });
 &lt;/script&gt;
 &lt;script type='text/javascript' src='https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'&gt;&lt;/script&gt;</header>
+          <!-- Required for Java 8u121 or later. See https://github.com/neo4j/neo4j-java-driver/pull/318  -->
+          <!--<additionalparam>&#45;&#45;allow-script-in-comments</additionalparam>-->
         </configuration>
         <executions>
           <execution>

--- a/src/net/i2p/crypto/eddsa/EdDSAEngine.java
+++ b/src/net/i2p/crypto/eddsa/EdDSAEngine.java
@@ -318,6 +318,8 @@ public final class EdDSAEngine extends Signature {
      *  sig = sign()
      *</pre>
      *
+     * @param data the message to be signed
+     * @return the signature
      * @throws SignatureException if update() already called
      * @see #ONE_SHOT_MODE
      */
@@ -336,6 +338,10 @@ public final class EdDSAEngine extends Signature {
      *  sig = sign()
      *</pre>
      *
+     * @param data byte array containing the message to be signed
+     * @param off the start of the message inside data
+     * @param len the length of the message
+     * @return the signature
      * @throws SignatureException if update() already called
      * @see #ONE_SHOT_MODE
      */
@@ -356,6 +362,9 @@ public final class EdDSAEngine extends Signature {
      *  ok = verify(signature)
      *</pre>
      *
+     * @param data the message that was signed
+     * @param signature of the message
+     * @return true if the signature is valid, false otherwise
      * @throws SignatureException if update() already called
      * @see #ONE_SHOT_MODE
      */
@@ -374,6 +383,11 @@ public final class EdDSAEngine extends Signature {
      *  ok = verify(signature)
      *</pre>
      *
+     * @param data byte array containing the message that was signed
+     * @param off the start of the message inside data
+     * @param len the length of the message
+     * @param signature of the message
+     * @return true if the signature is valid, false otherwise
      * @throws SignatureException if update() already called
      * @see #ONE_SHOT_MODE
      */
@@ -392,6 +406,11 @@ public final class EdDSAEngine extends Signature {
      *  ok = verify(signature, sigoff, siglen)
      *</pre>
      *
+     * @param data the message that was signed
+     * @param signature byte array containing the signature
+     * @param sigoff the start of the signature
+     * @param siglen the length of the signature
+     * @return true if the signature is valid, false otherwise
      * @throws SignatureException if update() already called
      * @see #ONE_SHOT_MODE
      */
@@ -410,6 +429,13 @@ public final class EdDSAEngine extends Signature {
      *  ok = verify(signature, sigoff, siglen)
      *</pre>
      *
+     * @param data byte array containing the message that was signed
+     * @param off the start of the message inside data
+     * @param len the length of the message
+     * @param signature byte array containing the signature
+     * @param sigoff the start of the signature
+     * @param siglen the length of the signature
+     * @return true if the signature is valid, false otherwise
      * @throws SignatureException if update() already called
      * @see #ONE_SHOT_MODE
      */

--- a/src/net/i2p/crypto/eddsa/EdDSAEngine.java
+++ b/src/net/i2p/crypto/eddsa/EdDSAEngine.java
@@ -435,7 +435,7 @@ public final class EdDSAEngine extends Signature {
     }
 
     /**
-     * @deprecated replaced with <a href="#engineSetParameter(java.security.spec.AlgorithmParameterSpec)">
+     * @deprecated
      */
     @Override
     protected void engineSetParameter(String param, Object value) {

--- a/src/net/i2p/crypto/eddsa/EdDSAKey.java
+++ b/src/net/i2p/crypto/eddsa/EdDSAKey.java
@@ -24,8 +24,8 @@ public interface EdDSAKey {
     String KEY_ALGORITHM = "EdDSA";
 
     /**
-     * return a parameter specification representing the EdDSA domain
-     * parameters for the key.
+     * @return a parameter specification representing the EdDSA domain
+     *         parameters for the key.
      */
     EdDSAParameterSpec getParams();
 }

--- a/src/net/i2p/crypto/eddsa/Utils.java
+++ b/src/net/i2p/crypto/eddsa/Utils.java
@@ -21,6 +21,8 @@ package net.i2p.crypto.eddsa;
 public class Utils {
     /**
      * Constant-time byte comparison.
+     * @param b a byte
+     * @param c a byte
      * @return 1 if b and c are equal, 0 otherwise.
      */
     public static int equal(int b, int c) {
@@ -34,6 +36,8 @@ public class Utils {
 
     /**
      * Constant-time byte[] comparison.
+     * @param b a byte[]
+     * @param c a byte[]
      * @return 1 if b and c are equal, 0 otherwise.
      */
     public static int equal(byte[] b, byte[] c) {

--- a/src/net/i2p/crypto/eddsa/math/Encoding.java
+++ b/src/net/i2p/crypto/eddsa/math/Encoding.java
@@ -28,6 +28,7 @@ public abstract class Encoding {
 
     /**
      * Encode a FieldElement in its $(b-1)$-bit encoding.
+     * @param x the FieldElement to encode
      * @return the $(b-1)$-bit encoding of this FieldElement.
      */
     public abstract byte[] encode(FieldElement x);
@@ -46,6 +47,7 @@ public abstract class Encoding {
      * than the $(b-1)$-bit encoding of -x. If $q$ is an odd prime and the encoding
      * is the little-endian representation of $\{0, 1,\dots, q-1\}$ then the negative
      * elements of $F_q$ are $\{1, 3, 5,\dots, q-2\}$.
+     * @param x the FieldElement to check
      * @return true if negative
      */
     public abstract boolean isNegative(FieldElement x);

--- a/src/net/i2p/crypto/eddsa/math/GroupElement.java
+++ b/src/net/i2p/crypto/eddsa/math/GroupElement.java
@@ -36,12 +36,13 @@ public class GroupElement implements Serializable {
 
     /**
      * Available representations for a group element.
-     * <p><ul>
+     * <ul>
      * <li>P2: Projective representation $(X:Y:Z)$ satisfying $x=X/Z, y=Y/Z$.
      * <li>P3: Extended projective representation $(X:Y:Z:T)$ satisfying $x=X/Z, y=Y/Z, XY=ZT$.
      * <li>P1P1: Completed representation $((X:Z), (Y:T))$ satisfying $x=X/Z, y=Y/T$.
      * <li>PRECOMP: Precomputed representation $(y+x, y-x, 2dxy)$.
      * <li>CACHED: Cached representation $(Y+X, Y-X, Z, 2dT)$
+     * </ul>
      */
     public enum Representation {
         /** Projective ($P^2$): $(X:Y:Z)$ satisfying $x=X/Z, y=Y/Z$ */
@@ -223,12 +224,13 @@ public class GroupElement implements Serializable {
      * <p>
      * A point $(x,y)$ is encoded by storing $y$ in bit 0 to bit 254 and the sign of $x$ in bit 255.
      * $x$ is recovered in the following way:
-     * <p><ul>
+     * </p><ul>
      * <li>$x = sign(x) * \sqrt{(y^2 - 1) / (d * y^2 + 1)} = sign(x) * \sqrt{u / v}$ with $u = y^2 - 1$ and $v = d * y^2 + 1$.
      * <li>Setting $β = (u * v^3) * (u * v^7)^{((q - 5) / 8)}$ one has $β^2 = \pm(u / v)$.
      * <li>If $v * β = -u$ multiply $β$ with $i=\sqrt{-1}$.
      * <li>Set $x := β$.
      * <li>If $sign(x) \ne$ bit 255 of $s$ then negate $x$.
+     * </ul>
      *
      * @param curve The curve.
      * @param s The encoded point.
@@ -864,7 +866,7 @@ public class GroupElement implements Serializable {
      * Constant time.
      * <p>
      * Preconditions: (TODO: Check this applies here)
-     *   $a[31] <= 127$
+     *   $a[31] \le 127$
      * @param a $= a[0]+256*a[1]+\dots+256^{31} a[31]$
      * @return the GroupElement
      */

--- a/src/net/i2p/crypto/eddsa/math/GroupElement.java
+++ b/src/net/i2p/crypto/eddsa/math/GroupElement.java
@@ -505,7 +505,7 @@ public class GroupElement implements Serializable {
      * $r$ in $P \times P$ representation:
      * <p>
      * $r = ((X' : Z'), (Y' : T'))$ where
-     * <p><ul>
+     * </p><ul>
      * <li>$X' = (X + Y)^2 - (Y^2 + X^2)$
      * <li>$Y' = Y^2 + X^2$
      * <li>$Z' = y^2 - X^2$
@@ -514,7 +514,7 @@ public class GroupElement implements Serializable {
      * $r$ converted from $P \times P$ to $P^2$ representation:
      * <p>
      * $r = (X'' : Y'' : Z'')$ where
-     * <p><ul>
+     * </p><ul>
      * <li>$X'' = X' * Z' = ((X + Y)^2 - Y^2 - X^2) * (2 * Z^2 - (y^2 - X^2))$
      * <li>$Y'' = Y' * T' = (Y^2 + X^2) * (2 * Z^2 - (y^2 - X^2))$
      * <li>$Z'' = Z' * T' = (y^2 - X^2) * (2 * Z^2 - (y^2 - X^2))$
@@ -647,14 +647,14 @@ public class GroupElement implements Serializable {
      * $r = p + q$ where $p = this = (X1 : Y1 : Z1 : T1), q = (q.X, q.Y, q.Z, q.T) = (Y2 + X2, Y2 - X2, Z2, 2 * d * T2)$
      * <p>
      * $r$ in $P \times P$ representation:
-     * <p><ul>
+     * </p><ul>
      * <li>$X' = (Y1 + X1) * (Y2 + X2) - (Y1 - X1) * (Y2 - X2)$
      * <li>$Y' = (Y1 + X1) * (Y2 + X2) + (Y1 - X1) * (Y2 - X2)$
      * <li>$Z' = 2 * Z1 * Z2 + 2 * d * T1 * T2$
      * <li>$T' = 2 * Z1 * T2 - 2 * d * T1 * T2$
      * </ul><p>
      * Setting $A = (Y1 - X1) * (Y2 - X2), B = (Y1 + X1) * (Y2 + X2), C = 2 * d * T1 * T2, D = 2 * Z1 * Z2$ we get
-     * <p><ul>
+     * </p><ul>
      * <li>$X' = (B - A)$
      * <li>$Y' = (B + A)$
      * <li>$Z' = (D + C)$

--- a/src/net/i2p/crypto/eddsa/math/ScalarOps.java
+++ b/src/net/i2p/crypto/eddsa/math/ScalarOps.java
@@ -18,16 +18,16 @@ public interface ScalarOps {
      * From the Ed25519 paper:<br>
      * Here we interpret $2b$-bit strings in little-endian form as integers in
      * $\{0, 1,..., 2^{(2b)}-1\}$.
-     * @param s
+     * @param s the scalar to reduce
      * @return $s \bmod l$
      */
     public byte[] reduce(byte[] s);
 
     /**
      * $r = (a * b + c) \bmod l$
-     * @param a
-     * @param b
-     * @param c
+     * @param a a scalar
+     * @param b a scalar
+     * @param c a scalar
      * @return $(a*b + c) \bmod l$
      */
     public byte[] multiplyAndAdd(byte[] a, byte[] b, byte[] c);

--- a/src/net/i2p/crypto/eddsa/math/bigint/BigIntegerLittleEndianEncoding.java
+++ b/src/net/i2p/crypto/eddsa/math/bigint/BigIntegerLittleEndianEncoding.java
@@ -39,6 +39,7 @@ public class BigIntegerLittleEndianEncoding extends Encoding implements Serializ
      *  Convert $x$ to little endian.
      *  Constant time.
      *
+     *  @param x the BigInteger value to encode
      *  @return array of length $b/8$
      *  @throws IllegalStateException if field not set
      */
@@ -75,6 +76,9 @@ public class BigIntegerLittleEndianEncoding extends Encoding implements Serializ
 
     /**
      *  Convert in to big endian
+     *
+     *  @param in the $(b-1)$-bit encoding of a FieldElement.
+     *  @return the decoded value as a BigInteger
      */
     public BigInteger toBigInteger(byte[] in) {
         byte[] out = new byte[in.length];

--- a/src/net/i2p/crypto/eddsa/math/ed25519/Ed25519FieldElement.java
+++ b/src/net/i2p/crypto/eddsa/math/ed25519/Ed25519FieldElement.java
@@ -62,13 +62,14 @@ public class Ed25519FieldElement extends FieldElement {
      * TODO-CR BR: $h$ is allocated via new, probably not a good idea. Do we need the copying into temp variables if we do that?
      * <p>
      * Preconditions:
-     * <p><ul>
+     * </p><ul>
      * <li>$|f|$ bounded by $1.1*2^{25},1.1*2^{24},1.1*2^{25},1.1*2^{24},$ etc.
      * <li>$|g|$ bounded by $1.1*2^{25},1.1*2^{24},1.1*2^{25},1.1*2^{24},$ etc.
      * </ul><p>
      * Postconditions:
-     * <p><ul>
+     * </p><ul>
      * <li>$|h|$ bounded by $1.1*2^{26},1.1*2^{25},1.1*2^{26},1.1*2^{25},$ etc.
+     * </ul>
      *
      * @param val The field element to add.
      * @return The field element this + val.
@@ -90,13 +91,14 @@ public class Ed25519FieldElement extends FieldElement {
      * TODO-CR BR: See above.
      * <p>
      * Preconditions:
-     * <p><ul>
+     * </p><ul>
      * <li>$|f|$ bounded by $1.1*2^{25},1.1*2^{24},1.1*2^{25},1.1*2^{24},$ etc.
      * <li>$|g|$ bounded by $1.1*2^{25},1.1*2^{24},1.1*2^{25},1.1*2^{24},$ etc.
      * </ul><p>
      * Postconditions:
-     * <p><ul>
+     * </p><ul>
      * <li>$|h|$ bounded by $1.1*2^{26},1.1*2^{25},1.1*2^{26},1.1*2^{25},$ etc.
+     * </ul>
      *
      * @param val The field element to subtract.
      * @return The field element this - val.
@@ -116,12 +118,13 @@ public class Ed25519FieldElement extends FieldElement {
      * TODO-CR BR: see above.
      * <p>
      * Preconditions:
-     * <p><ul>
+     * </p><ul>
      * <li>$|f|$ bounded by $1.1*2^{25},1.1*2^{24},1.1*2^{25},1.1*2^{24},$ etc.
      * </ul><p>
      * Postconditions:
-     * <p><ul>
+     * </p><ul>
      * <li>$|h|$ bounded by $1.1*2^{25},1.1*2^{24},1.1*2^{25},1.1*2^{24},$ etc.
+     * </ul>
      *
      * @return The field element (-1) * this.
      */
@@ -139,14 +142,14 @@ public class Ed25519FieldElement extends FieldElement {
      * Can overlap $h$ with $f$ or $g$.
      * <p>
      * Preconditions:
-     * <p><ul>
+     * </p><ul>
      * <li>$|f|$ bounded by
      * $1.65*2^{26},1.65*2^{25},1.65*2^{26},1.65*2^{25},$ etc.
      * <li>$|g|$ bounded by
      * $1.65*2^{26},1.65*2^{25},1.65*2^{26},1.65*2^{25},$ etc.
      * </ul><p>
      * Postconditions:
-     * <p><ul>
+     * </p><ul>
      * <li>$|h|$ bounded by
      * $1.01*2^{25},1.01*2^{24},1.01*2^{25},1.01*2^{24},$ etc.
      * </ul><p>
@@ -390,11 +393,11 @@ public class Ed25519FieldElement extends FieldElement {
      * Can overlap $h$ with $f$.
      * <p>
      * Preconditions:
-     * <p><ul>
+     * </p><ul>
      * <li>$|f|$ bounded by $1.65*2^{26},1.65*2^{25},1.65*2^{26},1.65*2^{25},$ etc.
      * </ul><p>
      * Postconditions:
-     * <p><ul>
+     * </p><ul>
      * <li>$|h|$ bounded by $1.01*2^{25},1.01*2^{24},1.01*2^{25},1.01*2^{24},$ etc.
      * </ul><p>
      * See {@link #multiply(FieldElement)} for discussion
@@ -546,11 +549,11 @@ public class Ed25519FieldElement extends FieldElement {
      * Can overlap $h$ with $f$.
      * <p>
      * Preconditions:
-     * <p><ul>
+     * </p><ul>
      * <li>$|f|$ bounded by $1.65*2^{26},1.65*2^{25},1.65*2^{26},1.65*2^{25},$ etc.
      * </ul><p>
      * Postconditions:
-     * <p><ul>
+     * </p><ul>
      * <li>$|h|$ bounded by $1.01*2^{25},1.01*2^{24},1.01*2^{25},1.01*2^{24},$ etc.
      * </ul><p>
      * See {@link #multiply(FieldElement)} for discussion

--- a/src/net/i2p/crypto/eddsa/math/ed25519/Ed25519LittleEndianEncoding.java
+++ b/src/net/i2p/crypto/eddsa/math/ed25519/Ed25519LittleEndianEncoding.java
@@ -26,15 +26,15 @@ public class Ed25519LittleEndianEncoding extends Encoding {
      * <li>Convert the field element to the 32 byte representation.
      * </ol><p>
      * The idea for the modulo $p$ reduction algorithm is as follows:
-     * <p>
+     * </p>
      * <h2>Assumption:</h2>
-     * <p><ul>
+     * <ul>
      * <li>$p = 2^{255} - 19$
      * <li>$h = h_0 + 2^{25} * h_1 + 2^{(26+25)} * h_2 + \dots + 2^{230} * h_9$ where $0 \le |h_i| \lt 2^{27}$ for all $i=0,\dots,9$.
      * <li>$h \cong r \mod p$, i.e. $h = r + q * p$ for some suitable $0 \le r \lt p$ and an integer $q$.
      * </ul><p>
      * Then $q = [2^{-255} * (h + 19 * 2^{-25} * h_9 + 1/2)]$ where $[x] = floor(x)$.
-     * <p>
+     * </p>
      * <h2>Proof:</h2>
      * <p>
      * We begin with some very raw estimation for the bounds of some expressions:
@@ -242,8 +242,9 @@ public class Ed25519LittleEndianEncoding extends Encoding {
      * Return false if $x$ is in $\{0,2,4,\dots,q-1\}$
      * <p>
      * Preconditions:
-     * <p><ul>
+     * </p><ul>
      * <li>$|x|$ bounded by $1.1*2^{26},1.1*2^{25},1.1*2^{26},1.1*2^{25}$, etc.
+     * </ul>
      *
      * @return true if $x$ is in $\{1,3,5,\dots,q-2\}$, false otherwise.
      */

--- a/src/net/i2p/crypto/eddsa/math/ed25519/Ed25519ScalarOps.java
+++ b/src/net/i2p/crypto/eddsa/math/ed25519/Ed25519ScalarOps.java
@@ -327,7 +327,7 @@ public class Ed25519ScalarOps implements ScalarOps {
      * $(ab+c) \bmod q$
      * <p>
      * Input:
-     * <p><ul>
+     * </p><ul>
      * <li>$a[0]+256*a[1]+\dots+256^{31}*a[31] = a$
      * <li>$b[0]+256*b[1]+\dots+256^{31}*b[31] = b$
      * <li>$c[0]+256*c[1]+\dots+256^{31}*c[31] = c$

--- a/src/net/i2p/crypto/eddsa/spec/EdDSAParameterSpec.java
+++ b/src/net/i2p/crypto/eddsa/spec/EdDSAParameterSpec.java
@@ -34,7 +34,11 @@ public class EdDSAParameterSpec implements AlgorithmParameterSpec, Serializable 
     private final GroupElement B;
 
     /**
-     *  @throws IllegalArgumentException if hash algorithm is unsupported or length is wrong
+     * @param curve the curve
+     * @param hashAlgo the JCA string for the hash algorithm
+     * @param sc the parameter L represented as ScalarOps
+     * @param B the parameter B
+     * @throws IllegalArgumentException if hash algorithm is unsupported or length is wrong
      */
     public EdDSAParameterSpec(Curve curve, String hashAlgo,
             ScalarOps sc, GroupElement B) {

--- a/src/net/i2p/crypto/eddsa/spec/EdDSAPrivateKeySpec.java
+++ b/src/net/i2p/crypto/eddsa/spec/EdDSAPrivateKeySpec.java
@@ -30,6 +30,8 @@ public class EdDSAPrivateKeySpec implements KeySpec {
     private final EdDSAParameterSpec spec;
 
     /**
+     *  @param seed the private key
+     *  @param spec the parameter specification for this key
      *  @throws IllegalArgumentException if seed length is wrong or hash algorithm is unsupported
      */
     public EdDSAPrivateKeySpec(byte[] seed, EdDSAParameterSpec spec) {
@@ -67,6 +69,7 @@ public class EdDSAPrivateKeySpec implements KeySpec {
      *  Initialize directly from the hash.
      *  getSeed() will return null if this constructor is used.
      *
+     *  @param spec the parameter specification for this key
      *  @param h the private key
      *  @throws IllegalArgumentException if hash length is wrong
      *  @since 0.1.1

--- a/src/net/i2p/crypto/eddsa/spec/EdDSAPublicKeySpec.java
+++ b/src/net/i2p/crypto/eddsa/spec/EdDSAPublicKeySpec.java
@@ -25,7 +25,9 @@ public class EdDSAPublicKeySpec implements KeySpec {
     private final EdDSAParameterSpec spec;
 
     /**
-     *  @throws IllegalArgumentException if key length is wrong
+     * @param pk the public key
+     * @param spec the parameter specification for this key
+     * @throws IllegalArgumentException if key length is wrong
      */
     public EdDSAPublicKeySpec(byte[] pk, EdDSAParameterSpec spec) {
         if (pk.length != spec.getCurve().getField().getb()/8)

